libs/wallet/provider/uphold/uphold.go

```diff
@@ -115,12 +116,33 @@ func init() {
 	} else {
 		proxy = nil
 	}
-	client = &http.Client{
+
+	fingerprintDialer := pindialer.MakeContextDialer(upholdCertFingerprint)
+
+	// Uphold reports HTTP 401 error when connecting with HTTP2, so disable
+	// HTTP/2 via setting TLSNextProto to an empty map. As defaultHTTPClient
+	// sets DialTLSContext and that also disables HTTP/2 unless
+	// ForceAttemptHTTP2 is set, we do not need to set TLSNextProto there, but
+	// we do it for clarity.
+	disableHTTP2 := make(
+		map[string]func(authority string, c *tls.Conn) http.RoundTripper,
+		0,
+	)
+	defaultHTTPClient = &http.Client{
 		Timeout: time.Second * 60,
 		Transport: middleware.InstrumentRoundTripper(
 			&http.Transport{
+				DialTLSContext: fingerprintDialer,
 				Proxy:          proxy,
-				DialTLSContext: pindialer.MakeContextDialer(upholdCertFingerprint),
+				TLSNextProto:   disableHTTP2,
+			}, "uphold"),
+	}
+	httpClientNoFingerprinting = &http.Client{
+		Timeout: defaultHTTPClient.Timeout,
```

Comment:
By this, a dependency between the two clients is established during initialisation.

A better option would be to declare the timeout as a constant, and then use it for both as opposed to coupling the instantiation of `httpClientNoFingerprinting` to that of `defaultHTTPClient`.

Also, consider more concise naming. E.g. `httpClientNoFP` could do just as well.

Additionally, it seems cleaner to have two separate maps for `TLSNextProto` as opposed to the proposed another global variable.

Reply:
Only once submitted the comment, I realised this is happening inside `init()` so technically there is no much to depend on in terms of initialisation. So you might as well disregard this comment.

Reply:
I renamed the variable as suggested.